File: src/main/java/xyz/chunkstories/api/exceptions/content/MeshLoadException.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.exceptions.content;

import xyz.chunkstories.api.content.Asset;

public class MeshLoadException extends AssetException {

	private static final long serialVersionUID = 5322485540396142553L;

	public MeshLoadException(Asset asset) {
		super(asset);
	}

	@Override
	public String getMessage() {
		return "Mesh from asset " + getAsset() + " failed to load.";
	}

}

File: src/main/java/xyz/chunkstories/api/exceptions/content/MeshLoadException.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.exceptions.content

import xyz.chunkstories.api.content.Asset

class MeshLoadException(asset: Asset) : AssetException(asset) {

    override val message: String?
        get() = "Mesh from asset $asset failed to load."

    companion object {

        private val serialVersionUID = 5322485540396142553L
    }

}
